emulators/mame/Portfile

```diff
@@ -66,20 +163,327 @@ build.args-append   ARCHOPTS="${configure.cxxflags}" \
+
+    # Paths to executables
+    dict set g_mame_python_info_dict py_bin \
+        "${prefix}/bin/python${p_ver_major}.${p_ver_minor}"
+    dict set g_mame_python_info_dict sphinx_bin \
+        "${prefix}/bin/sphinx-build-${p_ver_major}.${p_ver_minor}"
+
+    ui_debug "mame_python_populate_info: g_mame_python_info_dict: ${g_mame_python_info_dict}"
+
+    return 0
+}
+
+# Update dependencies, based on global Python info dictionary.
+# Note: Proc 'mame_python_populate_info' must be called first!
+proc mame_python_add_deps {} {
+    global g_mame_python_info_dict
+
+    set ver_major [dict get ${g_mame_python_info_dict} ver_major]
+    set ver_minor [dict get ${g_mame_python_info_dict} ver_minor]
+
+    depends_build-append \
+        "port:python${ver_major}${ver_minor}" \
+        "port:py${ver_major}${ver_minor}-sphinx" \
+        "port:py${ver_major}${ver_minor}-sphinxcontrib-svg2pdfconverter"
+
+    return 0
 }
 
+#------------------------------------------------------------------------------
+# Procs - High Level - Variant Support
+#------------------------------------------------------------------------------
+
+proc mame_variant_tools_setup {} {
+    build.args-append \
+        "TOOLS=1"
+
+    return 0
+}
+
+# Wrapper for python* variants to call, limiting copy-and-paste mayhem.
+proc mame_variant_python_setup {p_ver_major p_ver_minor} {
+    mame_python_populate_info ${p_ver_major} ${p_ver_minor}
+    mame_python_add_deps
+
+    return 0
+}
+
+#------------------------------------------------------------------------------
+# Procs - High Level - Phase Support
+#------------------------------------------------------------------------------
+
+proc mame_build_run {p_work_path p_target p_build_args_list} {
+    global build.cmd
+    global build.jobs
+    global build.pre_args
+
+    ui_debug "mame_build_run: p_work_path: ${p_work_path}"
+    ui_debug "mame_build_run: p_target: ${p_target}"
+    ui_debug "mame_build_run: p_build_args_list: ${p_build_args_list}"
+
+    set build_cmd_line \
+        "${build.cmd} \
+            --jobs=${build.jobs} \
+            [join ${build.pre_args}] \
+            ${p_target} \
+            [join ${p_build_args_list}]"
+
+    ui_debug "mame_build_run: build_cmd_line: ${build_cmd_line}"
+
+    system -W \
+        ${p_work_path} \
+        ${build_cmd_line}
+
+    return 0
+}
+
+#------------------------------------------------------------------------------
+# Variants
+#------------------------------------------------------------------------------
+
+variant tools \
+        description {Compile and install the mame tools like chdman and others} {
+    mame_variant_tools_setup
+}
+
+variant python36 \
+        description {Use python 3.6 for build} \
+        conflicts python37 python38 python39 {
+    mame_variant_python_setup 3 6
+}
+
+variant python37 \
+        description {Use python 3.7 for build} \
+        conflicts python36 python38 python39 {
+    mame_variant_python_setup 3 7
+}
```

Comment:
To simplify things, please remove these old python version variants. Only the newer pythons are needed. Honestly I would remove them all and simply always use python 3.8.

Reply:
Added this to ticket 61739, covering internal cleanup:

https://trac.macports.org/ticket/61739